w3d_translator/w3d_translator.py

```diff
@@ -1,10 +1,101 @@
+import typer
+import glob
+import threading
+import os
+
 from unity import UNITY_VERSION, UNITY_PATH
+from validate import validate_input  # , validate_output
+from threads import projectThread
+
+# TODO: Configure typer character length (100)
+# TODO: Configure color/style for typer.echo
+# TODO: Prompt user for confirmation if --force is used (?)
+
+# TEMPORARY OVERRIDES (for development)
+# Skip validate_output
+# Skip create_project and copy_files (xml_to_unity.py)
+# Only translate run.xml (xml_to_unity.py)
+
+
+# Opening message
+def greeting(in_dir, out_dir):
+    typer.echo("W3D TRANSLATOR")
+    typer.echo(f"Unity Version:\t {UNITY_VERSION}")
+    typer.echo(f'Please ensure Unity is installed at {UNITY_PATH}')
+    typer.echo(f"IN_DIR:\t\t {in_dir}")
+    typer.echo(f"OUT_DIR:\t {out_dir}")
+    typer.echo()
+
+
+# Ending message
+def farewell():
+    typer.echo()
+    typer.echo("Translation Complete!")
+
+
+def main(
+    in_dir: str = typer.Argument(
+        ...,
```

Comment:
what do the `...` do?

Reply:
None exactly sure how python interprets in but the end result is `typer` makes the CLI argument required